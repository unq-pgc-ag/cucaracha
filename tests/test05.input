
fun main() {
  a := True 
  b := False 
  c := True 
  x := a and b
  x := b or a
  x := not a
  x := a and b and c
  x := a and (b and c)
  x := a or b or c
  x := a or (b or c)
  x := a and b or c
  x := a and (b or c)
  x := a or b and c
  x := a or (b and c)
  x := not a and b
  x := not (a and b)
  x := a and not b
  x := not a or not b
  x := not (a or not b)
  x := not not a or not not b
  x := not (not a or not not b)
  x := not not (a or not not b)
  x := not (not not a or not b)
}
