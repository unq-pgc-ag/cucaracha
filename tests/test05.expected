(Program
  (Function
    main
    Unit
    (Block
      (StmtAssign
        a
        (ExprConstBool
          True
        )
      )
      (StmtAssign
        b
        (ExprConstBool
          False
        )
      )
      (StmtAssign
        c
        (ExprConstBool
          True
        )
      )
      (StmtAssign
        x
        (ExprAnd
          (ExprVar
            a
          )
          (ExprVar
            b
          )
        )
      )
      (StmtAssign
        x
        (ExprOr
          (ExprVar
            b
          )
          (ExprVar
            a
          )
        )
      )
      (StmtAssign
        x
        (ExprNot
          (ExprVar
            a
          )
        )
      )
      (StmtAssign
        x
        (ExprAnd
          (ExprAnd
            (ExprVar
              a
            )
            (ExprVar
              b
            )
          )
          (ExprVar
            c
          )
        )
      )
      (StmtAssign
        x
        (ExprAnd
          (ExprVar
            a
          )
          (ExprAnd
            (ExprVar
              b
            )
            (ExprVar
              c
            )
          )
        )
      )
      (StmtAssign
        x
        (ExprOr
          (ExprOr
            (ExprVar
              a
            )
            (ExprVar
              b
            )
          )
          (ExprVar
            c
          )
        )
      )
      (StmtAssign
        x
        (ExprOr
          (ExprVar
            a
          )
          (ExprOr
            (ExprVar
              b
            )
            (ExprVar
              c
            )
          )
        )
      )
      (StmtAssign
        x
        (ExprOr
          (ExprAnd
            (ExprVar
              a
            )
            (ExprVar
              b
            )
          )
          (ExprVar
            c
          )
        )
      )
      (StmtAssign
        x
        (ExprAnd
          (ExprVar
            a
          )
          (ExprOr
            (ExprVar
              b
            )
            (ExprVar
              c
            )
          )
        )
      )
      (StmtAssign
        x
        (ExprAnd
          (ExprOr
            (ExprVar
              a
            )
            (ExprVar
              b
            )
          )
          (ExprVar
            c
          )
        )
      )
      (StmtAssign
        x
        (ExprOr
          (ExprVar
            a
          )
          (ExprAnd
            (ExprVar
              b
            )
            (ExprVar
              c
            )
          )
        )
      )
      (StmtAssign
        x
        (ExprAnd
          (ExprNot
            (ExprVar
              a
            )
          )
          (ExprVar
            b
          )
        )
      )
      (StmtAssign
        x
        (ExprNot
          (ExprAnd
            (ExprVar
              a
            )
            (ExprVar
              b
            )
          )
        )
      )
      (StmtAssign
        x
        (ExprAnd
          (ExprVar
            a
          )
          (ExprNot
            (ExprVar
              b
            )
          )
        )
      )
      (StmtAssign
        x
        (ExprOr
          (ExprNot
            (ExprVar
              a
            )
          )
          (ExprNot
            (ExprVar
              b
            )
          )
        )
      )
      (StmtAssign
        x
        (ExprNot
          (ExprOr
            (ExprVar
              a
            )
            (ExprNot
              (ExprVar
                b
              )
            )
          )
        )
      )
      (StmtAssign
        x
        (ExprOr
          (ExprNot
            (ExprNot
              (ExprVar
                a
              )
            )
          )
          (ExprNot
            (ExprNot
              (ExprVar
                b
              )
            )
          )
        )
      )
      (StmtAssign
        x
        (ExprNot
          (ExprOr
            (ExprNot
              (ExprVar
                a
              )
            )
            (ExprNot
              (ExprNot
                (ExprVar
                  b
                )
              )
            )
          )
        )
      )
      (StmtAssign
        x
        (ExprNot
          (ExprNot
            (ExprOr
              (ExprVar
                a
              )
              (ExprNot
                (ExprNot
                  (ExprVar
                    b
                  )
                )
              )
            )
          )
        )
      )
      (StmtAssign
        x
        (ExprNot
          (ExprOr
            (ExprNot
              (ExprNot
                (ExprVar
                  a
                )
              )
            )
            (ExprNot
              (ExprVar
                b
              )
            )
          )
        )
      )
    )
  )
)
